web_sdk/golden_comparator/lib/goldens.dart

```diff
@@ -214,45 +222,60 @@ class _GoldensRepoFetcher {
 
     print('Fetching $_repository@$_revision');
 
-    if (!environment.webUiGoldensRepositoryDirectory.existsSync()) {
-      environment.webUiGoldensRepositoryDirectory.createSync(recursive: true);
-      await runProcess(
-        'git',
+    if (!_webUiGoldensRepositoryDirectory.existsSync()) {
+      _webUiGoldensRepositoryDirectory.createSync(recursive: true);
+      await _runGit(
         <String>['init'],
-        workingDirectory: environment.webUiGoldensRepositoryDirectory.path,
-        mustSucceed: true,
+        _webUiGoldensRepositoryDirectory.path,
       );
-      await runProcess(
-        'git',
+
+      await _runGit(
         <String>['remote', 'add', 'origin', _repository],
-        workingDirectory: environment.webUiGoldensRepositoryDirectory.path,
-        mustSucceed: true,
+        _webUiGoldensRepositoryDirectory.path,
       );
     }
 
-    await runProcess(
-      'git',
+    await _runGit(
       <String>['fetch', 'origin', 'master'],
-      workingDirectory: environment.webUiGoldensRepositoryDirectory.path,
-      mustSucceed: true,
+      _webUiGoldensRepositoryDirectory.path,
     );
-    await runProcess(
-      'git',
+    await _runGit(
       <String>['checkout', _revision],
-      workingDirectory: environment.webUiGoldensRepositoryDirectory.path,
-      mustSucceed: true,
+      _webUiGoldensRepositoryDirectory.path,
     );
   }
 
   Future<String> _getLocalRevision() async {
-    final io.File head = io.File(path.join(
-      environment.webUiGoldensRepositoryDirectory.path, '.git', 'HEAD'
-    ));
+    final io.File head = io.File(
+        path.join(_webUiGoldensRepositoryDirectory.path, '.git', 'HEAD'));
 
     if (!head.existsSync()) {
       return null;
     }
 
     return head.readAsStringSync().trim();
   }
+
+  /// Runs `git` with given arguments.
+  Future<int> _runGit(
+    List<String> arguments,
+    String workingDirectory,
+  ) async {
+    final io.Process process = await io.Process.start(
+      'git',
+      arguments,
+      workingDirectory: workingDirectory,
+      // Running the process in a system shell for Windows. Otherwise
+      // the process is not able to get Dart from path.
+      runInShell: io.Platform.isWindows,
+      mode: io.ProcessStartMode.inheritStdio,
+    );
+    final int exitCode = await process.exitCode;
+    if (exitCode != 0) {
+      throw Exception('Git command failed with arguments $arguments on '
+          'workingDirectory: $workingDirectory resulting with exitCode: '
+          '$exitCode');
+    }
+    return exitCode;
```

Comment:
Do we need to return the exit code if the value is always 0. I'd make it `Future<void>`.

Reply:
done